ocaml/xapi/xapi_host.ml

```diff
@@ -820,21 +820,23 @@ let is_slave ~__context ~host:_ = not (Pool_role.is_master ())
 let ask_host_if_it_is_a_slave ~__context ~host =
   let ask_and_warn_when_slow ~__context =
     let local_fn = is_slave ~host in
-    let timeout = 10. in
+    let timeout = Mtime.Span.(10 * s) in
     let task_name = Context.get_task_id __context |> Ref.string_of in
     let ip, uuid =
       ( Db.Host.get_address ~__context ~self:host
       , Db.Host.get_uuid ~__context ~self:host
       )
     in
     let rec log_host_slow_to_respond timeout () =
+      let minimum a b = if Mtime.Span.compare a b < 0 then b else a in
```

Comment:
I believe this is wrong. if `compare a b` < 0 then a < b, hence a is the minimum.

Reply:
Yeah, we want to return a unless b is a larger span of time.